js/views/conversation_view.js

```diff
@@ -327,8 +344,14 @@
             this.model.messageCollection.add(message, {merge: true});
             message.setToExpire();
 
+            // if the last seen indicator is old enough, it will go away.
+            // if it isn't, we want to make sure it's up to date
+            if (this.lastSeenIndicator) {
+                this.lastSeenIndicator.increment(1);
```

Comment:
I'm still leery of this increment strategy but at this point I think we'd have to migrate to tracking seen state on individual messages to avoid it.